source/linux/epoll_event_loop.c

```diff
@@ -545,7 +546,7 @@ static void s_main_loop(void *args) {
     struct epoll_loop *epoll_loop = event_loop->impl_data;
 
     /* set thread id to the thread of the event loop */
-    aws_atomic_store_int(&epoll_loop->thread_id, (size_t)aws_thread_current_thread_id());
+    aws_atomic_store_ptr(&epoll_loop->thread_id, &epoll_loop->thread.thread_id);
```

Comment:
I'm confused by having something called `x.thread_id` and something else called `x.thread.thread_id`. What's the meaning of the two

Reply:
typically nested structs indicate something that needs a memory barrier.

Reply:
The proposal: rename the fields
```
struct aws_thread thread_created_on;
struct aws_thread_id_t thread_currently_running_on;
```

Reply:
`x.thread` is a `struct aws_thread`. He's directly accessing its `thread_id` member instead of using `aws_thread_get_id(&epoll_loop->thread)`, which is a *bit* dirty because we don't typically access the members of our structs directly.

`x.thread_id` is an atomic that **points** to `x.thread.thread_id`. We could get rid of the atomic `x.thread_id` if we held a lock when querying `aws_thread_get_id(&epoll_loop->thread)`. But the way the atomics are set up here, it's a sneaky efficiency trick that I'm pretty sure works.

We could have **another** variable to avoid accessing `x.thread.thread_id` directly.
```
struct epoll_loop {
    ...
    struct aws_thread thread;
    aws_thread_id_t thread_id; // copy of thread's thread_id
    struct aws_atomic_var loop_thread_id_ptr; // points to thread_id
    ...
};
```

Reply:
Variable names should be clearer now.

@JonathanHenson: right, this is perhaps not as clean as it could be. I'm peeking inside a struct member. I think we're okay doing this (from a thread-safety pov) but will double-check. We could do as @graebm suggests and have a separate variable which we initialize with `aws_thread_current_thread_id()`.